lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,483 @@
+                type='str',
+                required=True
+            ),
+            server_name=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            collation=dict(
+                type='str',
+                required=False
+            ),
+            create_mode=dict(
+                type='str',
+                required=False
+            ),
+            source_database_id=dict(
+                type='str',
+                required=False
+            ),
+            source_database_deletion_date=dict(
+                type='datetime',
+                required=False
+            ),
+            restore_point_in_time=dict(
+                type='datetime',
+                required=False
+            ),
+            recovery_services_recovery_point_resource_id=dict(
+                type='str',
+                required=False
+            ),
+            edition=dict(
+                type='str',
+                required=False
+            ),
+            max_size_bytes=dict(
+                type='str',
+                required=False
+            ),
+            requested_service_objective_id=dict(
+                type='str',
+                required=False
+            ),
+            requested_service_objective_name=dict(
+                type='str',
+                required=False
+            ),
+            elastic_pool_name=dict(
+                type='str',
+                required=False
+            ),
+            read_scale=dict(
+                type='str',
+                required=False
+            ),
+            sample_name=dict(
+                type='str',
+                required=False
+            ),
+            zone_redundant=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.server_name = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDatabases, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                               supports_check_mode=True,
+                                               supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
```

Comment:
You're not doing anything with these values so instead of putting in all these if statements just change it to

```
for key in list(self.module_arg_spec.keys()) + ['tags']:
    setattr(self, key, kwargs[key])
```

You would need to make sure you set `self.key = None` in the `__init__` constructor above but this is the common way Azure modules are done.